### PR TITLE
removed promises from XRWebGLLayerFactory and XRMediaLayerFactory

### PR DESCRIPTION
**Pull request by cabanier**

### CODE REVIEW

**Comment by cabanier** [comment]
This will resolve #32

File: webxrlayers-1.bs [hunk 1]
```diff
@@ -492,18 +492,18 @@ interface XRWebGLLayerFactory {
...
-                                       XRLayerInit init);
-  Promise&lt;XRCylinderLayer&gt; requestCylinderLayer(XRTextureType textureType,
-                                               XRLayerInit init);
-  Promise&lt;XREquirectLayer&gt; requestEquirectLayer(XRTextureType textureType,
-                                               XRLayerInit init);
-  Promise&lt;XRCubeLayer&gt; requestCubeLayer(XRLayerInit init);
-
-  XRWebGLSubImage? getSubImage(XRLayer layer, XRFrame frame);
-  XRWebGLSubImage? getViewSubImage(XRLayer layer, XRView view);
+  XRProjectionLayer requestProjectionLayer(XRTextureType textureType,
```
Comment (toji) [inline webxrlayers-1.bs:495]: Ooh! One late comment! We've been using the term "request" exclusively with methods that return Promises till now. I wonder if this change means these should be called "create____Layer" instead?

**Comment by cabanier** [comment]
> LGTM to remove the Promises. I question whether or not throwing here is the right pattern, as it seems like it could catch some developers by surprise, but I'm really not sure what the common pattern is across other web APIs for this kind of behavior.

Yes. I was wondering about that as well. One nice thing about exceptions is that we can give more information but we could also log the reason in the console.

> I'm gonna reach out to some web ergonomics folks on the Google side to get some more information, but I'm fine with this landing in the meantime because the delta between throwing an exception and returning null is smaller (from an algorithm standpoint) than returning a promise vs not.

thanks!

File: webxrlayers-1.bs (hunk 1 above)
Comment (cabanier) [reply webxrlayers-1.bs:495]: I like "create" better than "request". 
I will make that change.

**Pull request merged**